polynote-kernel/src/main/scala/polynote/config/package.scala

```diff
@@ -12,13 +14,15 @@ package object config {
   sealed trait RepositoryConfig
   final case class ivy(
     base: String,
-    artifactPatternOpt: Option[String] = None,
-    metadataPatternOpt: Option[String] = None,
+    @JsonKey("artifact_pattern") artifactPatternOpt: Option[String] = None,
```

Comment:
Note that this also affects the JSON representation used in the ipynb file – if you have notebooks where you've set the patterns to deviate from the default, the keys will have to be updated in the ipynb file.